### PR TITLE
chore(internal): update trace writer debug log to include the status of the writer

### PR DESCRIPTION
**Pull request by brettlangdon**
In some scenarios, when using gevent, the trace writer thread can get in a state where it doesn't get restarted post fork.

This typically leads to a scenario where we keep trying to buffer spans as they finish, but because we are not flushing them, the buffer will continually overflow.

This change adds the status of the writer into the warning logs for when the buffer is full to be able to diagnose better if the writer is actively running or not.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.

### CODE REVIEW

**Review by Yun-Kim — DISMISSED** [review]

File: ddtrace/internal/writer.py [hunk 1]
```diff
@@ -555,11 +555,12 @@ def write(self, spans=None):
         except BufferFull as e:
             payload_size = e.args[0]
             log.warning(
-                "trace buffer (%s traces %db/%db) cannot fit trace of size %db, dropping",
+                "trace buffer (%s traces %db/%db) cannot fit trace of size %db, dropping. writer status: %s",
```
Comment (P403n1x87) [inline ddtrace/internal/writer.py:558]:
```suggestion
                "trace buffer (%s traces %db/%db) cannot fit trace of size %db, dropping (writer status: %s)",
```
or capital `Writer`?

Comment (brettlangdon) [reply ddtrace/internal/writer.py:558]: 🤷🏻 no strong preference. what do you like?

Comment (P403n1x87) [reply ddtrace/internal/writer.py:558]: I like the current code suggestion with the `(...)` more 🍭

Comment (brettlangdon) [reply ddtrace/internal/writer.py]: done.

**Review by P403n1x87 — APPROVED** [review]

**Review by ZStriker19 — APPROVED** [review]

**Pull request merged**